cpp/src/Slice/Preprocessor.cpp

```diff
@@ -252,398 +239,19 @@ Slice::Preprocessor::preprocess(bool keepComments, const string& languageArg)
-
-    //
-    // We now need to massage the result to get the desired output.
-    // First remove the backslash used to escape new lines.
-    //
-    string::size_type pos;
-    while ((pos = unprocessed.find(" \\\n")) != string::npos)
-    {
-        unprocessed.replace(pos, 3, "\n");
-    }
-
-    //
-    // Split filenames in separate lines:
-    //
-    // /foo/A.ice /foo/B.ice becomes
-    // /foo/A.ice
-    // /foo/B.ice
-    //
-    // C:\foo\A.ice C:\foo\B.ice becomes
-    // C:\foo\A.ice
-    // C:\foo\B.ice
-    //
-    pos = 0;
-#ifdef _WIN32
-    while ((pos = unprocessed.find(".ice ", pos)) != string::npos)
-    {
-        if (unprocessed.find(":", pos) == pos + 6)
-        {
-            unprocessed.replace(pos, 5, ".ice\n");
-            pos += 5;
-        }
-    }
-#else
-    while ((pos = unprocessed.find(".ice /", pos)) != string::npos)
-    {
-        unprocessed.replace(pos, 5, ".ice\n");
-    }
-#endif
-
-    //
-    // Get the main output file name.
-    //
-#ifdef _MSC_VER
-    string suffix = ".obj:";
-#else
-    string suffix = ".o:";
-#endif
-    pos = unprocessed.find(suffix) + suffix.size();
-    string result;
-    if (lang != SliceXML)
-    {
-        result = unprocessed.substr(0, pos);
-    }
-
-    vector<string> fullIncludePaths;
-    fullIncludePaths.reserve(includePaths.size());
-    for (const auto& includePath : includePaths)
-    {
-        fullIncludePaths.push_back(fullPath(includePath));
-    }
-
-    //
-    // Process each dependency.
-    //
-    string sourceFile;
-    vector<string> dependencies;
-
-    string::size_type end;
-    while ((end = unprocessed.find('\n', pos)) != string::npos)
-    {
-        end += 1;
-        string file = IceInternal::trim(unprocessed.substr(pos, end - pos));
-        if (file.rfind(".ice") == file.size() - 4)
-        {
-            if (IceInternal::isAbsolutePath(file))
-            {
-                if (file == _fileName)
-                {
-                    file = _shortFileName;
-                }
-                else
-                {
-                    //
-                    // Transform back full paths generated by mcpp to paths relative to the specified
-                    // include paths.
-                    //
-                    string newFile = file;
-                    for (auto p = fullIncludePaths.begin(); p != fullIncludePaths.end(); ++p)
-                    {
-                        if (file.compare(0, p->length(), *p) == 0)
-                        {
-                            string s = includePaths[static_cast<size_t>(p - fullIncludePaths.begin())] +
-                                       file.substr(p->length());
-                            if (IceInternal::isAbsolutePath(newFile) || s.size() < newFile.size())
-                            {
-                                newFile = s;
-                            }
-                        }
-                    }
-                    file = newFile;
```

Comment:
I removed this logic in the new implementation because I think it is bogus.

Here we shorten file names, by resolving them with the given include paths and settle for the one that produces the shortest path.

But when Make is processing the dependency files, it doesn't know about this include paths. I think this is also true for other tools.